xbmc-xrandr.c

```diff
@@ -597,7 +597,8 @@ static void set_transform(
   dest->filter = strdup(filter);
   dest->nparams = nparams;
   dest->params = malloc(nparams * sizeof(XFixed));
-  memcpy(dest->params, params, nparams * sizeof(XFixed));
+  if (nparams)
+    memcpy(dest->params, params, nparams * sizeof(XFixed));
```

Comment:
Calling `malloc` with size 0 is implementation defined. Better change it to this for predictable behavior:
```cpp
static void set_transform(
    transform_t* dest, XTransform* transform, char* filter, XFixed* params, int nparams)
{
  dest->transform = *transform;
  dest->filter = strdup(filter);
  dest->nparams = nparams;
  if (nparams)
  {
    dest->params = malloc(nparams * sizeof(XFixed));
    memcpy(dest->params, params, nparams * sizeof(XFixed));
  }
  else
    dest->params = NULL;
}
```